rebuild generated om/map classes before running tests

jwong-dayspring opened the pull request:
~~These files are generated by `propel:build` or `propel:model:build` and don't need to be part of the bundle.~~
The om/map files are necessary for first-run in projects that use LoginBundle, so we'll keep those.

This adds a step to the unit tests to generate the om/map classes. 
Set the `build.properties` to match PropelBundle's default for `propel.defaultTimeStampFormat` from `AbstractCommand`

jwong-dayspring committed:
remove generated om/map classes

dposner-dayspring reviewed (dismissed):
@jwong-dayspring 
there's a comment for build.properties:
'Set the build.properties to match PropelBundle's default for propel.defaultTimeStampFormat from AbstractCommand'
but the propel.defaultTimeStampFormat is not being set (line 9)?

jwong-dayspring committed:
restore om/map classes so installing bundle doesn't throw class not found error

jwong-dayspring committed:
revert date diff changes since they're not necessary with fix to default return type

jwong-dayspring commented:
> @jwong-dayspring there's a comment for build.properties: 'Set the build.properties to match PropelBundle's default for propel.defaultTimeStampFormat from AbstractCommand' but the propel.defaultTimeStampFormat is not being set (line 9)?

PropelBundle's AbstractCommand changes some of the default values for the Propel generator properties, and one of them is to change `propel.defaultTimeStampFormat` from a string formatted as `Y-m-d H:i:s` to blank, which is interpreted as `null`
See https://github.com/dayspring-tech/PropelBundle/blob/1.8/Resources/doc/propel_configuration.markdown
and https://github.com/dayspring-tech/PropelBundle/blob/1.8/Command/AbstractCommand.php#L669-L682

jwong-dayspring commented:
I was able to figure out that you must have run vendor/bin/propel-gen om to get clean copies of the om and map files. 
WebTestCase used to run the propel:build command during test setup, but that broke during some of the PropelBundle updates. That resulted it not getting the correct default timestamp format because PropelBundle changes the setting for defaultTimeStampFormat when you run bin/console propel:build , but running propel-gen om uses a different default format. 
I changed the build.properties for propel-gen om to match how PropelBundle behaves, and added propel-gen om to the composer test script.
